packages/contracts-bedrock/scripts/Deploy.s.sol

```diff
@@ -102,6 +115,18 @@ contract Deploy is Deployer {
         return keccak256(bytes(vm.envOr("IMPL_SALT", string("ethers phoenix"))));
     }
 
+    /// @notice Modifier that wraps a function with statediff recording.
+    ///         The returned AccountAccess[] array is then written to
+    ///         the `statediff-<name>.json` output file.
+    modifier statediff() {
+        vm.startStateDiffRecording();
+        _;
+        VmSafe.AccountAccess[] memory accesses = vm.stopAndReturnStateDiff();
+        string memory json = LibCheat.encodeAccountAccesses(accesses);
+        string memory statediffPath = string.concat("statediff-", name(), ".json");
```

Comment:
I'd suggest putting this in a snapshot folder to match other snapshot plans and clarify the intent, e.g. `snapshot/state-diff/<name>.json`